### ISSUE TITLE
AIO busted. FH5 Anti cheating update made some drastic changes.

### ISSUE THREAD

**Issue by ItsLogic** [Yeethan69/AIO#119]
AIO is broken for now until we get it figured out. Any issues relating to this will be closed and redirected here.

The devs added "Cheating Tool Detection" which makes the game crash when Cheat Engine, x64DBG, Process Hacker 2 or IDA are opened, there may be some other apps but ive only seen it happen with them so far. Custom built tools like the AIO are fine as well as custom Cheat Engine builds.

They also added memory integrity checks in some parts. The game will just crash if you edit a point they dont want you to which means the whole autoshow/garage tab is currently non functional along with some other features from cool shit like noclip.

TLDR memory integrity bad, Dont expect a quick update

**Comment by LegitSoulja** [comment]
I guess this for FH5, and not FH4 right?

**Comment by trqm** [comment]
How long does it take to download the new update

**Comment by trqm** [comment]
How long does it take to download the new update

**Comment by Arin-426** [comment]
> How long does it take to download the new update

You can download the update from cs rin, it's 8GB and currently working for online fix only, I.e offline cracks won't work/crash the game.

**Comment by trqm** [comment]
Where can I download it?

**Comment by Arin-426** [comment]
> Where can I download it?

cs rin ru, scroll to the bottom and sort posts by descending, you will see the links by InsaneRamzes

**Comment by trqm** [comment]
Where can I download it?

**Comment by trqm** [comment]
????

**Comment by Bubbly0** [comment]
interesting so u guys are working on a way to bypass it?

**Comment by trqm** [comment]
I downloaded how do I work

**Comment by trqm** [comment]
Can it be used anymore?
@ItsLogic

**Comment by Yeethan69** [comment]
As of now, the AIO does not work. Do not ask about it, asking will not make it update any faster. 
Also stop asking about game cracks, they have nothing to do with the AIO and we will not help with such topic

**Comment by trqm** [comment]
Well, we will go to solve this

**Comment by LegitSoulja** [comment]
> > I am new here and this might be a dumb (and irrelevant question) but what are custom builds of CE, and where/how can I download?
> 
> > 
> 
> > Edit- So the existing cheat tables and CE will be useless now and never work again? Or will there be a fix posted for this?
> 
> 
> 
> You can build cheat engine from source while removing all references to the words "cheat engine" and it will not force the game to close as you can see from this image: 
> 
> ![image](https://user-images.githubusercontent.com/38233332/200899903-0f97ac0c-4802-46f4-afc8-a32c2a62c9b7.png)
> 
> 
> 
> Some parts of cheat tables and the AIO will still work. Current builds of cheat engine wont work unless someone bypasses the "kill list" (personally I dont really care about the kill list so it will be the last thing I look into)
> 
> 
> 
> 
> 
> > interesting so u guys are working on a way to bypass it?
> 
> 
> 
> yeah although dont keep your hopes up. They really dont want you messing with the game anymore

Personally, I use LunarEngine. All references to "Cheat" is removed. There's still other cheat engines that works like ArtMoney for instance..

I wish I had much reverse engineering experience to help out.

**Comment by LegitSoulja** [comment]
> > > > I am new here and this might be a dumb (and irrelevant question) but what are custom builds of CE, and where/how can I download?
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > Edit- So the existing cheat tables and CE will be useless now and never work again? Or will there be a fix posted for this?
> > > 
> > > 
> > > You can build cheat engine from source while removing all references to the words "cheat engine" and it will not force the game to close as you can see from this image:
> > > ![image](https://user-images.githubusercontent.com/38233332/200899903-0f97ac0c-4802-46f4-afc8-a32c2a62c9b7.png)
> > > Some parts of cheat tables and the AIO will still work. Current builds of cheat engine wont work unless someone bypasses the "kill list" (personally I dont really care about the kill list so it will be the last thing I look into)
> > > > interesting so u guys are working on a way to bypass it?
> > > 
> > > 
> > > yeah although dont keep your hopes up. They really dont want you messing with the game anymore
> > 
> > 
> > Personally, I use LunarEngine. All references to "Cheat" is removed. There's still other cheat engines that works like ArtMoney for instance..
> > As far as CheatTables they would need to be converted. I actually had a repo that does this for LunarEngine but removed it. I might re-upload it back.
> > I wish I had much reverse engineering experience to help out.
> 
> Thanks, I'll try it out. Can you re upload that repo?

Had to remake it as I lost original files, but here you go. I tested a few tables and it works. Hopefully there's nothing I missed https://legitsoulja.github.io/CT2LT/

**Comment by Arin-426** [comment]
> > > > > I am new here and this might be a dumb (and irrelevant question) but what are custom builds of CE, and where/how can I download?
> > > > 
> > > > 
> > > > > 
> > > > 
> > > > 
> > > > > Edit- So the existing cheat tables and CE will be useless now and never work again? Or will there be a fix posted for this?
> > > > 
> > > > 
> > > > You can build cheat engine from source while removing all references to the words "cheat engine" and it will not force the game to close as you can see from this image:
> > > > ![image](https://user-images.githubusercontent.com/38233332/200899903-0f97ac0c-4802-46f4-afc8-a32c2a62c9b7.png)
> > > > Some parts of cheat tables and the AIO will still work. Current builds of cheat engine wont work unless someone bypasses the "kill list" (personally I dont really care about the kill list so it will be the last thing I look into)
> > > > > interesting so u guys are working on a way to bypass it?
> > > > 
> > > > 
> > > > yeah although dont keep your hopes up. They really dont want you messing with the game anymore
> > > 
> > > 
> > > Personally, I use LunarEngine. All references to "Cheat" is removed. There's still other cheat engines that works like ArtMoney for instance..
> > > As far as CheatTables they would need to be converted. I actually had a repo that does this for LunarEngine but removed it. I might re-upload it back.
> > > I wish I had much reverse engineering experience to help out.
> > 
> > 
> > Thanks, I'll try it out. Can you re upload that repo?
> 
> Had to remake it as I lost original files, but here you go. I tested a few tables and it works. Hopefully there's nothing I missed https://legitsoulja.github.io/CT2LT/

Thanks a lot, I'll try it out tomorrow.

**Comment by gigamash77** [comment]
> AIO is broken for now until we get it figured out. Any issues relating to this will be closed and redirected here.
> 
> The devs added "Cheating Tool Detection" which makes the game crash when Cheat Engine, x64DBG, Process Hacker 2 or IDA are opened, there may be some other apps but ive only seen it happen with them so far. Custom built tools like the AIO are fine as well as custom Cheat Engine builds.
> 
> They also added memory integrity checks in some parts. The game will just crash if you edit a point they dont want you to which means the whole autoshow/garage tab is currently non functional along with some other features from cool shit like noclip.
> 
> TLDR memory integrity bad, Dont expect a quick update

ok, good to know, i wouldn't have expected anti cheat after such a long time. thanks for the information.

**Comment by LegitSoulja** [comment]
@Arin-426 good shit. 
@gigamash77 I bet you retards who uses AIO is the reason they did.

**Comment by gigamash77** [comment]
> @Arin-426 good shit. @gigamash77 I bet you retards who uses AIO is the reason they did.

this is about technical understanding and the exchange of changeable options. maybe you don't know what github is.

**Comment by Bubbly0** [comment]
i managed to keep stand for FH5 open for 2mins before it crashed didnt show all the cars tho does that mean something probably not but idkkk

**Comment by LegitSoulja** [comment]
> > @Arin-426 good shit. @gigamash77 I bet you retards who uses AIO is the reason they did.
> 
> this is about technical understanding and the exchange of changeable options. maybe you don't know what github is. we are not kiddis here who want to cheat. we juggle numbers to see what's possible. you should broaden your horizons.

Apparently you don’t know what Github is, you have no repository and no experience with any programming languages. I’m not a kid, but you wonder why they implemented anti cheat? As I said, the retards who use this tool using speed, anti gravity, flying hacks etc. Doesn’t have anything to do with Github. If this tool was private, there wouldn’t have been any anti cheat implemented

**Comment by Arin-426** [comment]
> > > @Arin-426 good shit. @gigamash77 I bet you retards who uses AIO is the reason they did.
> > 
> > 
> > this is about technical understanding and the exchange of changeable options. maybe you don't know what github is. we are not kiddis here who want to cheat. we juggle numbers to see what's possible. you should broaden your horizons.
> 
> Apparently you don’t know what Github is, you have no repositories and no experience with any programming languages. I’m not a kid, but you wonder why they implemented anti cheat? As you’re here, obviously you cheat. As I said, the retards who use this tool with speed, anti gravity, flying hacks is probably the reason why anti cheat measures was implemented. Doesn’t have anything to do with Github. If this tool was private, there wouldn’t have been any anti cheat implemented

Before I say anything, I want to make this clear I dont use this tool or any other to cheat like that or in Online mode, just wanna collect the cars lol Anyways, your converter worked just fine but I updated the game today and as soon as I open Lunar Engine, the game crashes, that means the game can detect LE too. Any solution to this?

**Comment by LegitSoulja** [comment]
> > > > @Arin-426 good shit. @gigamash77 I bet you retards who uses AIO is the reason they did.
> > > 
> > > 
> > > this is about technical understanding and the exchange of changeable options. maybe you don't know what github is. we are not kiddis here who want to cheat. we juggle numbers to see what's possible. you should broaden your horizons.
> > 
> > 
> > Apparently you don’t know what Github is, you have no repositories and no experience with any programming languages. I’m not a kid, but you wonder why they implemented anti cheat? As you’re here, obviously you cheat. As I said, the retards who use this tool with speed, anti gravity, flying hacks is probably the reason why anti cheat measures was implemented. Doesn’t have anything to do with Github. If this tool was private, there wouldn’t have been any anti cheat implemented
> 
> Before I say anything, I want to make this clear I dont use this tool or any other to cheat like that or in Online mode, just wanna collect the cars lol Anyways, your converter worked just fine but I updated the game today and as soon as I open Lunar Engine, the game crashes, that means the game can detect LE too. Any solution to this?

I apologize for my rant, it wasn't directly towards anyone, just those who use this tool in online modes to take advantage of others specifically. 

In regards to [LunarEngine](https://github.com/manovisible/lunarengine), are you attaching a debugger or does it crash as soon as you select the process? LE should work, unless "Engine" is also a keyword detected and not only "Cheat". 

Also do you have discord?

**Comment by Arin-426** [comment]
> > > > > @Arin-426 good shit. @gigamash77 I bet you retards who uses AIO is the reason they did.
> > > > 
> > > > 
> > > > this is about technical understanding and the exchange of changeable options. maybe you don't know what github is. we are not kiddis here who want to cheat. we juggle numbers to see what's possible. you should broaden your horizons.
> > > 
> > > 
> > > Apparently you don’t know what Github is, you have no repositories and no experience with any programming languages. I’m not a kid, but you wonder why they implemented anti cheat? As you’re here, obviously you cheat. As I said, the retards who use this tool with speed, anti gravity, flying hacks is probably the reason why anti cheat measures was implemented. Doesn’t have anything to do with Github. If this tool was private, there wouldn’t have been any anti cheat implemented
> > 
> > 
> > Before I say anything, I want to make this clear I dont use this tool or any other to cheat like that or in Online mode, just wanna collect the cars lol Anyways, your converter worked just fine but I updated the game today and as soon as I open Lunar Engine, the game crashes, that means the game can detect LE too. Any solution to this?
> 
> I apologize for my rant, it wasn't directly towards anyone, just those who use this tool in online modes to take advantage of others specifically.
> 
> In regards to [LunarEngine](https://github.com/manovisible/lunarengine), are you attaching a debugger or does it crash as soon as you select the process? LE should work, unless "Engine" is also a keyword detected and not only "Cheat". I could possibly do a custom build of CheatEngine again with as much remove that I think may be detected and update the cheat tables to work with it.
> 
> By chance, do you have discord?

It crashes even before I can select the process. Like if the game is open and I Alt Tab and open CE/LE the crashes within seconds and if I open CE/LE before starting the game, the game won't start at all unless I close CE/LE.  

And yes, I have Discord (Arin#3452). I think we should continue over there? This is getting a bit out of topic I think lol

**Comment by Bubbly0** [comment]
i managed to open stand and make it show all the cars guys what i did was i opened up the stand menu then i went and clicked on FH5 and as soon as it started loading i went back to stand and clicked on "shown" real quick and Forza continued to load and didnt crash it worked and showed all the cars the game didnt crash this time

**Comment by xiaoshuai1688** [comment]
![QQ截图20221116181343](https://user-images.githubusercontent.com/118350581/202153263-63c2c7dc-bb03-4f2f-a65c-b834bee2fc6b.png)
I first used the CE to perform a scan of the FH5 auto show, and then used the CT to activate all the cars and enter the auto show before the scan was over. FH5 didn't crash. But repeat the steps every time you buy a car.

**Comment by hanjiayou2010** [comment]
Steam has recently updated Horizon 5, resulting in the inability to attach to the game

**Comment by LegitSoulja** [comment]
> Steam has recently updated Horizon 5, resulting in the inability to attach to the game,so you're going to give up?

I'm sure it's still possible, maybe just not user mode.

**Comment by d5aqoep** [comment]
The car ID method still works to add any car from car collections screen. But you still need to know the exact value to change.

**Comment by ItsLogic** [comment]
> Steam has recently updated Horizon 5, resulting in the inability to attach to the game,so you're going to give up?

no

**Comment by P13rlU** [comment]
> > I am new here and this might be a dumb (and irrelevant question) but what are custom builds of CE, and where/how can I download?
> > Edit- So the existing cheat tables and CE will be useless now and never work again? Or will there be a fix posted for this?
> 
> You can build cheat engine from source while removing all references to the words "cheat engine" and it will not force the game to close as you can see from this image: ![image](https://user-images.githubusercontent.com/38233332/200899903-0f97ac0c-4802-46f4-afc8-a32c2a62c9b7.png)
> 
> Some parts of cheat tables and the AIO will still work. Current builds of cheat engine wont work unless someone bypasses the "kill list" (personally I dont really care about the kill list so it will be the last thing I look into)
> 
> > interesting so u guys are working on a way to bypass it?
> 
> yeah although dont keep your hopes up. They really dont want you messing with the game anymore

can you share your version of cheat engine, i can't find it anywhere, and when i try to use lunar engine, the game still crashes

**Comment by LegitSoulja** [comment]
> > > I am new here and this might be a dumb (and irrelevant question) but what are custom builds of CE, and where/how can I download?
> > > Edit- So the existing cheat tables and CE will be useless now and never work again? Or will there be a fix posted for this?
> > 
> > 
> > You can build cheat engine from source while removing all references to the words "cheat engine" and it will not force the game to close as you can see from this image: ![image](https://user-images.githubusercontent.com/38233332/200899903-0f97ac0c-4802-46f4-afc8-a32c2a62c9b7.png)
> > Some parts of cheat tables and the AIO will still work. Current builds of cheat engine wont work unless someone bypasses the "kill list" (personally I dont really care about the kill list so it will be the last thing I look into)
> > > interesting so u guys are working on a way to bypass it?
> > 
> > 
> > yeah although dont keep your hopes up. They really dont want you messing with the game anymore
> 
> can you share your version of cheat engine, i can't find it anywhere, and when i try to use lunar engine, the game still crashes

you can find it in their discord server

**Comment by ItsLogic** [comment]
New working AIO now out. Full memory integiry bypass

**Comment by Harskyll** [comment]
> > > > I am new here and this might be a dumb (and irrelevant question) but what are custom builds of CE, and where/how can I download?
> > > > Edit- So the existing cheat tables and CE will be useless now and never work again? Or will there be a fix posted for this?
> > > 
> > > 
> > > You can build cheat engine from source while removing all references to the words "cheat engine" and it will not force the game to close as you can see from this image: ![image](https://user-images.githubusercontent.com/38233332/200899903-0f97ac0c-4802-46f4-afc8-a32c2a62c9b7.png)
> > > Some parts of cheat tables and the AIO will still work. Current builds of cheat engine wont work unless someone bypasses the "kill list" (personally I dont really care about the kill list so it will be the last thing I look into)
> > > > interesting so u guys are working on a way to bypass it?
> > > 
> > > 
> > > yeah although dont keep your hopes up. They really dont want you messing with the game anymore
> > 
> > 
> > can you share your version of cheat engine, i can't find it anywhere, and when i try to use lunar engine, the game still crashes
> 
> you can find it in their discord server

Could u send me their discord ?
